src/raw.rs

```diff
@@ -208,83 +240,110 @@ pub const REDISMODULE_HASH_DELETE: *const RedisModuleString = 1 as *const RedisM
+    redis_call!(raw RedisModule_CallReplyBool(reply)) != 0
 }
 
 /// # Panics
 ///
 /// Panics if the Redis server doesn't support replying with bool (since RESP3).
 #[allow(clippy::not_unsafe_ptr_arg_deref)]
 pub fn call_reply_double(reply: *mut RedisModuleCallReply) -> f64 {
-    unsafe { RedisModule_CallReplyDouble.unwrap()(reply) }
+    redis_call!(RedisModule_CallReplyDouble(reply))
 }
 
 /// # Panics
 ///
 /// Panics if the Redis server doesn't support replying with bool (since RESP3).
 #[allow(clippy::not_unsafe_ptr_arg_deref)]
 pub fn call_reply_big_number(reply: *mut RedisModuleCallReply) -> Option<String> {
-    unsafe {
-        let mut len: size_t = 0;
-        let reply_string: *mut u8 =
-            RedisModule_CallReplyBigNumber.unwrap()(reply, &mut len) as *mut u8;
+    let mut len: size_t = 0;
+    String::from_utf8({
+        let reply_string =
+            redis_call!(raw RedisModule_CallReplyBigNumber(reply, &mut len)) as *mut u8;
         if reply_string.is_null() {
             return None;
         }
-        String::from_utf8(slice::from_raw_parts(reply_string, len).to_vec()).ok()
-    }
+        unsafe { slice::from_raw_parts(reply_string, len).to_vec() }
+    })
+    .ok()
 }
 
 /// # Panics
 ///
 /// Panics if the Redis server doesn't support replying with bool (since RESP3).
 #[allow(clippy::not_unsafe_ptr_arg_deref)]
 pub fn call_reply_verbatim_string(reply: *mut RedisModuleCallReply) -> Option<(String, Vec<u8>)> {
-    unsafe {
-        let mut len: size_t = 0;
-        let format: *const u8 = ptr::null();
-        let reply_string: *mut u8 =
-            RedisModule_CallReplyVerbatim.unwrap()(reply, &mut len, &mut (format as *const c_char))
-                as *mut u8;
-        if reply_string.is_null() {
-            return None;
-        }
-        Some((
+    let mut len: size_t = 0;
+    let format: *const u8 = ptr::null();
+    let reply_string = redis_call!(raw RedisModule_CallReplyVerbatim(reply, &mut len, &mut (format as *const c_char)))
+        as *mut u8;
+    if reply_string.is_null() {
+        return None;
+    }
+    Some(unsafe {
+        (
             String::from_utf8(slice::from_raw_parts(format, 3).to_vec()).ok()?,
             slice::from_raw_parts(reply_string, len).to_vec(),
-        ))
-    }
+        )
+    })
+}
+
+/// Aborts the invocation of the blocking commands.
+#[allow(clippy::not_unsafe_ptr_arg_deref)]
+pub fn call_reply_promise_abort(
+    reply: *mut RedisModuleCallReply,
+    private_data: *mut *mut ::std::os::raw::c_void,
+) -> Status {
+    redis_call!(RedisModule_CallReplyPromiseAbort(reply, private_data))
+}
+
+macro_rules! generate_transparent_binding {
+    ($(#[$outer:meta])* $vis:vis $rust_name:ident => $raw_function:ident, $ret_type:ty, $(($arg:ident, $typ:ty)),*) => {
+        $(#[$outer])*
+        #[allow(clippy::not_unsafe_ptr_arg_deref)]
+        $vis fn $rust_name($($arg: $typ),*) -> $ret_type {
+            raw_call!($raw_function, $($arg),*)
+        }
+    };
```

Comment:
Discussion: This is one of the ways we could potentially leverage to simplify writing one-to-one function calls (one-liners), which perform some basic processing. I thought this wasn't elegant, and the best way would be to have a procedural macro instead. An example of the invocation:

```rust
generate_transparent_binding!(
    /// # Panics
    ///
    /// Panics if the Redis server doesn't support replying with bool (since RESP3).
    pub call_reply_set_element => RedisModule_CallReplySetElement,
    *mut RedisModuleCallReply,
    (reply, *mut RedisModuleCallReply),
    (idx, usize)
);
```

Reply:
I agree that procedural macro is better here. Can make all this glue code redudent.